packages/opa-react/src/authz-provider.tsx

```diff
@@ -72,9 +75,34 @@ export default function AuthzProvider({
   defaultInput,
   defaultFromResult,
 }: AuthzProviderProps) {
+  const defaultQueryFn = async ({
+    queryKey,
+    meta = {},
+    signal,
+  }: QueryFunctionContext) => {
+    const [path, input] = queryKey as any; // TODO(sr): tweak types
+    const fromResult = meta["fromResult"] as any; // TODO(sr): tweak types
+    return path
+      ? sdk.evaluate<Input, Result>(path, input, {
+          fromResult,
+          fetchOptions: { signal },
+        })
+      : sdk.evaluateDefault<Input, Result>(input, {
+          fromResult,
+          fetchOptions: { signal },
+        });
+  };
+  const queryClient = new QueryClient({
+    defaultOptions: {
+      queries: {
+        queryFn: defaultQueryFn,
+      },
+    },
+  });
+
   const context = useMemo<AuthzProviderContext>(
```

Comment:
ℹ️ Putting the `queryClient` into `context` _here_ is what makes our lib independent of other usages of `react-query` within the same (surrounding) application.

Reply:
I see, instead of using `QueryClientProvider` you integrated into AuthzContext.Provider, but the tradeoff is that QueryClient needs to be explicitly set to useQuery (from your usage below).

is it that much of a burden to simply wrap `QueryClientProvider`, so the downstream useQuery is more convenient to use? (and other React Query helpers can be used like useQueryClient)

```
<AuthzContext.Provider value={context}>
  <QueryClientProvider client={queryClient}>
    {children}
  </QueryClientProvider>
</AuthzContext.Provider>
```

Reply:
re-reading the PR description

> The queryClient is added to AuthzContext, so we're not trampling over any of our users' usage of react-query via QueryClientProvider.

Reply:
alternative (and maybe proper) to pass in a custom context when getting the queryClient instance instead of referring to the queryClient directly from the context (I wonder what are the implications)
https://tanstack.com/query/v4/docs/framework/react/reference/useQueryClient

> const queryClient = useQueryClient({ context })

Reply:
Is that still a thing? When clicking the "go to latest version" button, I'm sent here:https://tanstack.com/query/latest/docs/framework/react/reference/useQueryClient and I'm not quite sure how _that_ applies now. 😳 If you could give an example, I'd appreciate that.

Reply:
oh my bad, I didn't realize I was looking at an older version.  Don't mind what I said, your approach looks good